docs/advanced/types-react-ap.md

```diff
@@ -29,6 +29,7 @@ Not Commonly Used but Good to know
 
 - `Ref` - used to type `innerRef`
 - `ElementType` - used for higher order components or operations on components
+- `ReactElement` - [only used for `cloneElement`](https://www.reddit.com/r/reactjs/comments/ia8sdi/any_other_typescript_users_constantly_confused/g1npahe/) aka it's pretty rarely used
```

Comment:
```suggestion
- `ReactElement` - [can be used if you want to pass it to `cloneElement`](https://www.reddit.com/r/reactjs/comments/ia8sdi/any_other_typescript_users_constantly_confused/g1npahe/) aka it's pretty rarely used
```

`cloneElement` was only the first thing that came to mind. There are probably more use cases. Accessing properties on the element comes to mind but this is sketchy to begin with. At the point you do this you shouldn't need a cheatsheet. 

```
const element: ReactElement;
element.props;
element.type;